src/main/java/past/cwdt/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -63,4 +76,23 @@ public List<BookDto> search(BookDtoSearchByParameters searchParameters) {
                 .toList();
     }
 
+    @Override
+    public List<BookDtoWithoutCategoryIds> findBooksByCategory(Pageable pageable, Long id) {
+        return bookRepository.findAllByCategoryId(id).stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
+    }
+
+    private void checkIfBookExistsById(Long id) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't find book with id " + id);
+        }
+    }
+
+    private void setCategories(CreateBookRequestDto requestDto, Book book) {
```

Comment:
it is better to rework this method into:
```
private Set<Category> getCategoriesFrom(request)
```

setting categories inside of this method complicates readability